src/network/core/tcp_game.cpp

```diff
@@ -197,3 +200,17 @@ NetworkRecvStatus NetworkGameSocketHandler::Receive_SERVER_MOVE(Packet *p) { ret
 NetworkRecvStatus NetworkGameSocketHandler::Receive_CLIENT_MOVE(Packet *p) { return this->ReceiveInvalidPacket(PACKET_CLIENT_MOVE); }
 NetworkRecvStatus NetworkGameSocketHandler::Receive_SERVER_COMPANY_UPDATE(Packet *p) { return this->ReceiveInvalidPacket(PACKET_SERVER_COMPANY_UPDATE); }
 NetworkRecvStatus NetworkGameSocketHandler::Receive_SERVER_CONFIG_UPDATE(Packet *p) { return this->ReceiveInvalidPacket(PACKET_SERVER_CONFIG_UPDATE); }
+
+void NetworkGameSocketHandler::DeferDeletion()
+{
+	_deferred_deletions.push_back(this);
+	this->is_pending_deletion = true;
+}
+
+/* static */ void NetworkGameSocketHandler::ProcessDeferredDeletions()
+{
+	for (NetworkGameSocketHandler *cs : _deferred_deletions) {
+		delete cs;
+	}
+	_deferred_deletions.clear();
```

Comment:
Can't `_deferred_deletions` be a `std::vector<std::unique_ptr<NetworkGameSocketHandler>>` so this becomes just a `_deferred_deletions.clear()` without manually deleting objects?

Reply:
Wouldn't that be a bit confusing, as unique_ptr implies a different ownership model than what is actually used for the various socket handlers?

Reply:
Not sure if it's confusing. To me it looks like `_deferred_deletions` owns the handlers once they are passed into it, but if it confuses others then keeping it pointers that are manually `delete`d is fine by me as well.

Reply:
OK, I'll update the PR with this a bit later.